lib/puppet/network/http_pool.rb

```diff
@@ -28,6 +28,7 @@ def self.http_client_class=(klass)
   # @param verify_peer [Boolean] Whether to verify the peer credentials, if possible. Verification will not take place if the CA certificate is missing.
   # @return [Puppet::Network::HTTP::Connection]
   #
+  # @deprecated Use {#http_connection} instead.
```

Comment:
There is a `Puppet.deprecation_warning` which will log once per line of code, but I don't want to enable it right now, since everything in puppet is still using the deprecated methods. I think it's best to wait until after the 6.4 release goes out, and emit the deprecation in a future release, but well before 7. I'll add a ticket to do that.

Reply:
Filed as https://tickets.puppetlabs.com/browse/PUP-9513